### PR TITLE
[Instrumentation.Process] Removed CPU utilization metric `process.cpu.utilization`.

### PR DESCRIPTION
**Pull request by Yun-Ting**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/831.
Related to: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/948

## Changes

Used static ctor for ProcessMetrics and made Meter static.
Removed/updated CPU utilization related tests.
Added tests to check thread safety.

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #

### CODE REVIEW

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Review by xiang17 — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* [Instrumentation.Process] Removed CPU utilization metric.
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md:5]: We are in the Process changelog, so the tag is not needed.

```suggestion
* Removed CPU utilization metric.
```

**Review by Kielek — APPROVED** [review]
It is part of experimental ote-spec
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/process-metrics.md

it will be good to eventually bring it back.

**Comment by Kielek** [comment]
Please update changelog. It might be breaking change for part of the users.

File: src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md:5]: Please put the name of the metric being removed, so it is easier for users looking at the changelog to understand without having to open the PR.

File: test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs [hunk 2]
```diff
@@ -173,8 +212,6 @@ public void CheckValidGaugeValueWhen2MeterProviderInstancesHaveTheSameMeterName(
 
         meterProviderA.ForceFlush(MaxTimeToAllowForFlush);
 
-        // Note: This fails due to:
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs:176]: this note is removed intentionally?

**Review by alanwest — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs (hunk 2 above)
Comment (Yun-Ting) [reply test/OpenTelemetry.Instrumentation.Process.Tests/ProcessMetricsTests.cs:176]: yes, the test will not fail with the current design

File: src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md (hunk 1 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/CHANGELOG.md]: fixed; thank you

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@Yun-Ting Could you fix the conflicts, and we are good to merge

**Review by reyang — APPROVED** [review]

**Comment by utpilla** [comment]
Was the `MeterProvider` disposal issue the only reason to remove the `process.cpu.utilization`? We could have overcome that by keeping a count of the instrumentation instances and only dispose the `Meter` when there are no active `ProcessInstrumentation` instances?

**Pull request merged**